blog/2020/09/rr-memory-magic.md

```diff
@@ -0,0 +1,503 @@
+@def authors = "Keno Fischer"
+@def published = "23 September 2020"
+@def title = "Remotely Debugging Faulty Memory - an rr case study"
+@def rss_pubdate = Date(2020, 5, 2)
+@def rss = """Since the release of Julia 1.5 we've gotten lots of rr traces. This tells the story of a particularly interesting one."""
+
+~~~
+  <link rel="stylesheet" type="text/css" href="/assets/blog/2020-05-02-rr/asciinema-player.css" />
+  <script src="/assets/blog/2020-05-02-rr/asciinema-player.js"></script>
+~~~
+
+A [few months ago](../05/rr) on this blog, I introduced a new Julia feature that lets users easily submit rr traces along with their bug reports (please go read that post if you haven't, otherwise the rest of this post may not make much sense). Since Julia 1.5 was officially released a little over a month ago, more than a dozen users
+have made use of this capability to send us detailed bug reports. So far, everything went to plan.
+
+~~~
+<center>
+<img id="meme" src="/assets/blog/2020-09-23-rr-memory/rr-meme.jpeg"/>
+</center><br/>
+~~~
+
+In this post I want to tell the story of issue [#37524](https://github.com/JuliaLang/julia/issues/37524), the most unusual of the several rr traces we received. As spoiled in the title of this blog post, the eventual issue turned
+out to be faulty memory. However, by popular demand, and to dispel any rumors of the use of black magic, I thought
+it might be worth stepping through the analysis of this issue step by step.
+
+When debugging an rr trace, the first thing to do is usually to just replay it to the point of the crash to look around (sometimes the crash is obvious from the state of the program at the crash location, other times, more history is required). In this case the initial investigation was done by my colleague
+[Tim Besard](https://github.com/maleadt). Downloading the trace and starting the replay he would have seen the following. Note that this recording is sped up about 20x (it runs for about two minutes real time) and lightly edited at the end to focus on only the relevant output.
+
+~~~
+  <asciinema-player author="Keno Fischer"
```

Comment:
This player expands on the page when you press play which is slightly annoying but I don't know if there is anything to do about that.